src/pages/index.js

```diff
@@ -96,13 +96,15 @@ function handleEditProfileSubmit(formData) {
 }
 
 function handleEditProfilePictureSubmit(data) {
+  editProfilePicture.setSubmitText(true);
   api
     .updateProfilePicture(data.url)
-    .then(() => {
-      profilePictureImage.src = data.url;
+    .then(userInfo.setUserProfilePicture(data.url))
```

Comment:
In blocks `then`, `catch`, `finally` you need to pass only the **reference** to the function (not a function call) otherwise the function is called immediately without waiting for the response from the server. You can pass an **unnamed** arrow function as well.